lib/order.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'text'
+
+class Order
+  attr_reader :menu, :current, :cost
+
+  def initialize(menu = Takeaway::DISHES, text = Text)
+    @menu = menu
+    @current = Hash.new { |h, k| h[k] = 0 }
+    @cost = 0
+    @text = text
+  end
+
+  def add_to_order(dish, quantity)
+    fail "That isn't on the menu" unless menu.has_key?(dish.to_sym)
+    update_current(dish, quantity)
+    add_to_cost(dish, quantity)
+  end
+
+  def breakdown
+    fail no_items if @cost.zero?
+    order_breakdown
+  end
+
+  def pay(amount)
+    fail no_items if @cost.zero?
+    fail "Please pay £#{cost} to complete order" if amount != @cost
+    @text.send
+  end
+
+  private
+
+  def update_current(dish, quantity)
+    @current[dish.to_sym] += quantity
+    puts "#{quantity} #{dish} added to your order"
```

Comment:
This seems to be a bit of confusion between the view layer and the business logic.